### PR TITLE
[frozen] Add frozen library

### PR DESCRIPTION
**Pull request by theidexisted**
**Describe the pull request**

Add one new library frozen

- What does your PR fix? Fixes issue #
NA

- Which triplets are supported/not supported? Have you updated the CI baseline?
NA

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
YES

### CODE REVIEW

File: ports/frozen/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/frozen/portfile.cmake:1]: `include(vcpkg_common_functions)` is no longer needed.
Could you please remove this line?

File: ports/frozen/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    SHA512 b175a03b4a0263937e176675558e2df0bd4174799e2c5d7138842235fa24be57bccd8b96fddb5791e055b9de44211063f195142de73acf9d4f52a9a37f7055cc
+    HEAD_REF master
+)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+    OPTIONS
+    -DBUILD_TESTING=OFF
+    -Dfrozen.tests=OFF
```
Comment (NancyLi1013) [inline ports/frozen/portfile.cmake:16]: Could you please update the format?
```
OPTIONS
    -DBUILD_TESTING=OFF
    -Dfrozen.tests=OFF
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/frozen/portfile.cmake (hunk 2 above)
Comment (prehistoric-penguin) [reply ports/frozen/portfile.cmake]: I have fixed the configuration files, could you please review it again

**Review by NancyLi1013 — APPROVED** [review]

**Comment by prehistoric-penguin** [comment]
vcpkg-windows-PR-test

**Comment by NancyLi1013** [comment]
Thanks for your attention.
The regressions on Windows are not related with this PR.
So you don't need to care about this. We will try to rerun once solved.

File: ports/frozen/usage [hunk 3]
```diff
@@ -0,0 +1,4 @@
+The package frozen provides CMake integration:
+
+	find_package(frozen CONFIG REQUIRED)
+	target_link_libraries(main PRIVATE frozen::frozen-headers)
```
Comment (ras0219-msft) [inline ports/frozen/usage:4]: This should be automatically detected by our cmake heuristics -- why do we need a usage file here?

Comment (prehistoric-penguin) [reply ports/frozen/usage]: Thank you, I have removed this file.  BTW, is this file not needed for all libraries?

Comment (ras0219-msft) [reply ports/frozen/usage]: If the library generates `XYZ-config.cmake` files, we have some heuristics for automatically detecting this information. However, if those heuristics fail or the library relies on a built-in `FindXYZ` module, we need this usage file to fill in the gaps.

File: ports/frozen/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      -Dfrozen.tests=OFF
+)
+vcpkg_install_cmake()
+
+vcpkg_fixup_cmake_targets(CONFIG_PATH share/cmake/frozen TARGET_PATH share/frozen)
+
+file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/lib ${CURRENT_PACKAGES_DIR}/debug)
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+file(INSTALL ${CMAKE_CURRENT_LIST_DIR}/usage DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT})
```
Comment (NancyLi1013) [inline ports/frozen/portfile.cmake:23]: Please also remove this line, since it doesn't exist now.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**